docs: add timeout info to argocd-update step (#3995)
akuity/kargo: docs/docs/50-user-guide/60-reference-docs/30-promotion-steps/argocd-update.md

_Additionally, it is recommended that if a promotion process is expected to
sometimes encounter an active deny window, the `argocd-update` step should be
configured with a timeout that is at least as long as the longest expected deny

> window. (This timeout can be set using the [`retry.timeout`](../15-promotion-templates.md#step-retries) field. The default is five minutes)._

Reviewer: ```suggestion
window. The step's default timeout of five minutes can be overridden using thecan be [`retry.timeout`](../15-promotion-templates.md#step-retries) field.
```
